api/backend/multibackend.go

```diff
@@ -18,52 +18,28 @@ import (
 	"github.com/square/metrics/api"
 )
 
-type sequentialMultiBackend struct {
-	Backend api.Backend
+type ParallelTimeseriesStorageWrapper struct {
+	limit                int
+	tickets              chan struct{}
+	TimeseriesStorageAPI api.TimeseriesStorageAPI
 }
 
-func NewSequentialMultiBackend(backend api.Backend) api.MultiBackend {
-	return &sequentialMultiBackend{Backend: backend}
-}
-
-func (m *sequentialMultiBackend) FetchMultipleSeries(request api.FetchMultipleRequest) (api.SeriesList, error) {
-	series := make([]api.Timeseries, len(request.Metrics))
-	var err error = nil
-	for i, metric := range request.Metrics {
-		series[i], err = m.Backend.FetchSingleSeries(request.ToSingle(metric))
-		if err != nil {
-			return api.SeriesList{}, err
-		}
-	}
-
-	return api.SeriesList{
-		Series:    series,
-		Timerange: request.Timerange,
-	}, nil
-}
-
-type parallelMultiBackend struct {
-	limit   int
-	tickets chan struct{}
-	Backend api.Backend
-}
-
-func NewParallelMultiBackend(backend api.Backend, limit int) api.MultiBackend {
+func NewParallelMultiBackend(backend api.TimeseriesStorageAPI, limit int) *ParallelTimeseriesStorageWrapper {
```

Comment:
Yeah, individual variable renames were done where it was appropriate, but sometimes it was just bogging things down too much. After this change we can go through with a grep -Ri backend and change all these.